pkg/skaffold/runner/new.go

```diff
@@ -233,6 +236,107 @@ func getSyncer(cfg sync.Config) sync.Syncer {
 	return sync.NewSyncer(cfg)
 }
 
+/*
+The "default deployer" is used in `skaffold apply`, which uses a `kubectl` deployer to actuate resources
+on a cluster regardless of provided deployer configuration in the skaffold.yaml.
+The default deployer will honor a select set of deploy configuration from an existing skaffold.yaml:
+	- deploy.StatusCheckDeadlineSeconds
+	- deploy.Logs.Prefix
+	- deploy.Kubectl.Flags
+	- deploy.Kubectl.DefaultNamespace
+	- deploy.Kustomize.Flags
+	- deploy.Kustomize.DefaultNamespace
+For a multi-config project, we do not currently support resolving conflicts between differing sets of this deploy configuration.
+Therefore, in this function we do implicit validation of the provided configuration, and fail if any conflict cannot be resolved.
+*/
+func getDefaultDeployer(runCtx *runcontext.RunContext, labels map[string]string) (deploy.Deployer, error) {
+	deployCfgs := runCtx.DeployConfigs()
+
+	var kFlags *latest.KubectlFlags
+	var logPrefix string
+	var defaultNamespace *string
+	var kubeContext string
+	statusCheckTimeout := -1
+
+	for _, d := range deployCfgs {
+		if d.KubeContext != "" {
+			if kubeContext != "" {
```

Comment:
maybe you can relax this to allow all specified kubecontext be the same?
```suggestion
			if kubeContext != "" && kubeContext != d.KubeContext {
```

Reply:
ah yes thanks i meant to do this. should match the logic for the other checks here